soroban-env-host/src/host.rs

```diff
@@ -102,7 +108,7 @@ pub(crate) struct HostImpl {
     objects: RefCell<Vec<HostObject>>,
     storage: RefCell<Storage>,
     context: RefCell<Vec<Frame>>,
-    budget: RefCell<Budget>,
+    budget: Budget,
```

Comment:
This is a great point and thank you for the elaborate comment. 
By "it's not really possible to achieve" do you mean it is impractical/inadvisable doing so since there are multiple `Rc`<`RefMut` > holding on to the same budget? We *could* clone the underneath `budgetImpl` and create a new Budget from it, if we really wanted to?
Also the whole `deep_clone` is incompatible to budget metering, since it returns `-> Self ` not `Result<Self, HostError>`. So there isn't really any budget charging going on underneath currently. 
I guess my bigger question is, do we want budget charging for `deep_clone` at all? Under which scenarios is it used? Since it belongs to the `EnvBase`, is it only called by the guest(VM) in which case the metering is accounted for by the VM?

Reply:
Re: "not possible" yes, exactly, there are too many budget Rc pointers sprinkled all through the object graph. Theoretically possible; practically not really.

Re: budgeting deep_clone in general, don't worry about it. deep_clone exists for taking a snapshot of a host when doing certain types of testing, like exploring multiple paths in a fuzzer or model checker. It's not something users are expected to call in production (and the version in the user's Guest view just returns .. the same empty Guest structure, it's zero cost on the Guest, only meaningful on the Host)